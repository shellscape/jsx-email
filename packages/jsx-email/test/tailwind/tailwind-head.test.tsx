// @ts-ignore
import React from 'react';

import { jsxToString, render, Head, Html, Tailwind } from '../../dist';

const Component = () => (
  <Html>
    <Tailwind>
      <Head>
        <title>test title</title>
      </Head>
    </Tailwind>
  </Html>
);

describe('Tailwind + Head', async () => {
  beforeEach(() => {
    vi.restoreAllMocks();
    vi.resetModules();
  });

  test('should not clobber <Head> content', async () => {
    expect(await jsxToString(<Component />)).toMatchSnapshot();
    expect(await render(<Component />, { pretty: true })).toMatchSnapshot();
  });
});
