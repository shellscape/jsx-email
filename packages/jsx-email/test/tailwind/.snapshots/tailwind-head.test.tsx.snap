// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tailwind + Head > should not clobber <Head> content 1`] = `"<html lang=\\"en\\" dir=\\"ltr\\"><head><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\"/><title>test title</title></head><style tailwind>/* layer: preflights */</style></head></html>"`;

exports[`Tailwind + Head > should not clobber <Head> content 2`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\">
<html lang=\\"en\\" dir=\\"ltr\\">

  <head>
    <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
    <title>test title</title>
    <style tailwind>
      /* layer: preflights */
    </style>
  </head>

  <body></body>

</html>"
`;
